### PR TITLE
Black formatting of existing code and inclusion of formatting check in workflow

### PR DESCRIPTION
**Pull request by JuliaKukulies**

### CODE REVIEW

File: conda-requirements.txt
Comment (JuliaKukulies) [reply conda-requirements.txt:9]: You are right, I fixed this!

**Comment by freemansw1** [comment]
I think we should merge this into `dev` rather than `main`. Also, I'm confused as to why the docker check hasn't run yet...

**Comment by w-k-jones** [comment]
> 

Just changed the base to dev, thought that that might run the docker checks but it doesn't seem to

**Comment by freemansw1** [comment]
Now that it's passing the CI checks, I'm happy for this to be merged, but I'll let @w-k-jones as it's his review.

**Comment by fsenf** [comment]
Hi, I was just about to merge the PR, but then did the black reformatting by myself to be really on the safe side. Interestingly, I don't get exactly the same results. In my case, some more spaces have been added, e.g. 
 
```bash
> git diff fabians-PRcheck..juliasPR 
...

 def calculate_cog_untracked(mass, mask):
-    """ caluclate centre of gravity and mass for untracked parts of domain
+    """caluclate centre of gravity and mass for untracked parts of domain
     Input:
-    mass:       iris.cube.Cube 
+    mass:       iris.cube.Cube
...
```

Any idea what happend?

**Comment by fsenf** [comment]
@sean , @w-k-jones & @JuliaKukulies : If you don't think that these super-small differences matter, please press the merge button and step forward!

**Comment by JuliaKukulies** [comment]
Hmm..that is interesting. I used black 22.1.0, but tried to see if the same thing happens, when I use the same version as you, with:

`> conda list black 
# packages in environment at /home/juli/anaconda3/envs/blacktest:
#
# Name                    Version                   Build  Channel
black                     19.10b0                  py37_0    conda-forge
`
However, all files remain unchanged in my case. These are really small differences, but on the other hand the idea with black is to have everything exactly consistent. If you go down the [rabbit hole of formatting](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.htm)(section "empty lines"), I think it is against PEP-8  to have empty lines directly after the function definition, since you already have an indentation. Do you pass the formatting check when you push your changes?

**Comment by w-k-jones** [comment]
> Hi, I was just about to merge the PR, but then did the black reformatting by myself to be really on the safe side. Interestingly, I don't get exactly the same results. In my case, some more spaces have been added, e.g.
> 
> ```shell
> > git diff fabians-PRcheck..juliasPR 
> ...
> 
>  def calculate_cog_untracked(mass, mask):
> -    """ caluclate centre of gravity and mass for untracked parts of domain
> +    """caluclate centre of gravity and mass for untracked parts of domain
>      Input:
> -    mass:       iris.cube.Cube 
> +    mass:       iris.cube.Cube
> ...
> ```
> 
> Any idea what happend?

Not sure why your black formatting has included the extra spaces, however I believe that the PR is formatting correctly (without spaces) so I'm happy to merge if there are no other exceptions

**Review by w-k-jones — APPROVED** [review]

**Comment by fsenf** [comment]
OK, as a last check: My `black` version on my OS leaves all files in @JuliaKukulies PR unchanged, so we won't have any circular PRs in future ... Let's go forward ...

**Pull request merged**
